cmake/Corrosion.cmake

```diff
@@ -738,8 +738,11 @@ function(_add_cargo_build out_cargo_build_out_dir)
     set(cargo_target_linker $<$<BOOL:${linker}>:${cargo_target_linker_var}=${linker}>)
 
     if(Rust_CROSSCOMPILING AND (CMAKE_C_COMPILER_TARGET OR CMAKE_CXX_COMPILER_TARGET))
+        string(REPLACE "," "\$<COMMA>" c_opts_target "${CMAKE_C_COMPILE_OPTIONS_TARGET}")
```

Comment:
are there any cases where `CMAKE_<LANG>_COMPILE_OPTIONS_TARGET`  contains a comma?

Reply:
No, but if there ever was one, the CMake error would be pretty cryptic. :sweat_smile:

Learned this the hard way in another change I got lined up, which gives the same treatment to `--sysroot`. There, the QCC parameter is `-Wc,-isysroot,`.

It's probably fine to skip the escaping here though, if you prefer.

Reply:
> another change I got lined up, which gives the same treatment to --sysroot

In that case, could you perhaps follow-up in the upstream CMake issue https://gitlab.kitware.com/cmake/cmake/-/issues/21382#note_852685 and clarify if CMake can officially document these two variables? Brad king seems to have indicated this is fine in the linked comment, but it would be good to have some confirmation, and even better if we could merge a PR to CMake first which documents these variables (to make them part of the official API)

> It's probably fine to skip the escaping here though, if you prefer.

I would prefer skipping it unless necessary.

Reply:
Nice find, I'll approach upstream to get these documented.

Reply:
@mvf Did you receive any feedback from the CMake Maintainers?